test/hotspot/jtreg/runtime/malloctrace/MallocHooksTest.java

```diff
@@ -595,11 +597,25 @@ private static void testUniqueStacks() throws Exception {
                                   "-XX:MallocTraceStackDepth=14").start();
         checkIsAttachable(p);
         p.getInputStream().read();
-        OutputAnalyzer oa = callJcmd(p, "MallocTrace.dump", "-percentage=100");
+        // The output can be large, so dump it directly, since the output of
+        // jcmd is temporary hold in the JVM and there is a limit of 100 MB.
+        final String[] stacks = new String[1];
```

Comment:
Any reason why you are using a String[] here instead of a single String object?

Reply:
You cannot modify a local variable in an anonymous inner class. So the array trick is the common way.